ansible/roles/test/files/ptftests/vxlan_traffic.py

```diff
@@ -122,6 +135,32 @@ def read_ptf_macs(self):
 
         return addrs
 
+    def verify_all_addresses_used_equally(self, nhs, returned_ip_addresses):
+        '''
+           Verify the ECMP functionality using 2 checks.
+           Check 1 verifies every nexthop address has been used.
+           Check 2 verifies the distribution of number of packets among the nexthops.
+           Params: nhs: the nexthops that are configured.
+                   returned_ip_addresses: The dict containing the nh addresses and corresponding packet counts.
+        '''
+        # Check #1 : All addresses have been used.
+        if set(nhs) - set(returned_ip_addresses.keys()) == set([]):
+            logger.info("    Each address has been used")
+            logger.info("Packets sent:{} distribution:".format(self.packet_count))
+            for nh_address in returned_ip_addresses.keys():
+                logger.info("      {} : {}".format(nh_address, returned_ip_addresses[nh_address]))
+            # Check #2 : The packets are almost equally distributed.
+            # Every next-hop should have received within 1% of the packets that we sent per nexthop(which is self.packet_count).
+            # This check is valid only if there are large enough number of packets(300). Any lower number will need higher tolerance(more than 2%).
+            if self.packet_count > MINIMUM_PACKETS_FOR_ECMP_VALIDATION:
+                tolerance = 0.01
```

Comment:
@rraghav-cisco please clarify what this 1% value is based on. Unfortunately I was not able to find it in feature requirement docs